io/io/test/TFileMergerTests.cxx

```diff
@@ -221,3 +221,47 @@ TEST(TFileMerger, ChangeFile)
    gSystem->Unlink("file6640mergerinput_2.root");
    gSystem->Unlink("file6640mergeroutput.root");
 }
+
+TEST(TFileMerger, SelectiveMergeWithDirectories)
+{
+   constexpr auto input = "selectiveMerge_input.root";
+   constexpr auto output = "selectiveMerge_output.root";
+   {
+      TH1F histo("histo", "Histo", 2, 0, 1);
+      TFile infile(input, "recreate");
+      auto dir = infile.mkdir("A");
+      dir->WriteObject(&histo, "Histo_A");
+      dir = infile.mkdir("B");
+      dir->WriteObject(&histo, "Histo_B");
+      dir = infile.mkdir("C")->mkdir("D");
+      dir->WriteObject(&histo, "Histo_D");
+      dir = infile.mkdir("E")->mkdir("F");
+      dir->WriteObject(&histo, "Histo_F");
+   }
+
+   {
+      TFileMerger fileMerger(false);
+      fileMerger.AddFile(input);
```

Comment:
For completeness sake, consider using 2 files, selecting 3 histos in the subdirectory(ies) with one of the histos in both file, one of the histo in only the first file, one of the  histo only in the second file (in addition to all the histo that are not selected and can be in both files).

Reply:
This is a good extension, so I will add this soon-ish.

Reply:
Done!